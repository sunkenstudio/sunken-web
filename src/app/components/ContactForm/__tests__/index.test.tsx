"use client";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { ContactForm, ContactFormProps } from "..";
import {
  BrandFixture,
  ContactFixture,
  HeroFixture,
  SectionFixture,
} from "@/app/fixtures";
import * as Formik from "formik";

describe("ContactForm", () => {
  beforeAll(() => {
    const mockFieldChange = jest.fn();
    const mockSubmit = jest.fn();

    const useFormikContextMock = jest.spyOn(Formik, "useFormik");
    useFormikContextMock.mockReturnValue({
      initialValues: {},
      getFieldMeta: mockFieldChange,
      onSubmit: mockSubmit,
    } as unknown as any);
  });
  const renderComponent = (overrides: Partial<ContactFormProps> = {}) => {
    const defaults = (props: Partial<ContactFormProps>): ContactFormProps => ({
      hero: HeroFixture(),
      sections: [SectionFixture()],
      brand: BrandFixture(),
      contact: ContactFixture(),
      ...props,
    });
    return render(<ContactForm {...defaults(overrides)} />);
  };
  it("match snapshot", () => {
    const rendered = renderComponent();
    expect(rendered).toMatchSnapshot();
  });
});
