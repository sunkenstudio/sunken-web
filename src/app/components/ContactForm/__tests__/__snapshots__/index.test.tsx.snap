// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm All stories should render successfully Renders Primary story 1`] = `
<body
  class="chakra-ui-light"
>
  <div>
    <div
      class="css-9fal1s"
      id="contact"
    >
      <div
        class="css-1kpgg84"
      >
        <div
          class="css-ojqyia"
        >
          <img
            alt="Bg image - logo"
            class="chakra-image css-o0922y"
            src="https://sunkenstudio-strapi-cms.nyc3.digitaloceanspaces.com/b8c12199347804e795219ef484f5572e.jpeg"
            typename="ComponentCommonImage"
          />
          <div
            class="css-12lgfyx"
          />
        </div>
      </div>
      <div
        class="chakra-stack css-7pq1ey"
      >
        <h3
          class="chakra-heading css-1qtl148"
        >
          Contact
        </h3>
        <form
          style="display: flex; width: 100%; justify-content: center;"
        >
          <div
            class="chakra-stack css-ynastw"
          >
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <label
                class="chakra-form__label css-g6pte"
                for="name"
                id="field-:r0:-label"
              >
                NAME
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-1tfjd1n"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <input
                aria-required="true"
                class="chakra-input css-15izoeg"
                data-testid="input-name"
                id="name"
                name="NAME"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <label
                class="chakra-form__label css-g6pte"
                for="start_time"
                id="field-:r1:-label"
              >
                STARTTIME
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-1tfjd1n"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <input
                aria-required="true"
                class="chakra-input css-15izoeg"
                data-testid="input-start_time"
                id="start_time"
                name="STARTTIME"
                required=""
                type="time"
                value=""
              />
            </div>
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <label
                class="chakra-form__label css-g6pte"
                for="date"
                id="field-:r2:-label"
              >
                DATE
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-1tfjd1n"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <input
                aria-required="true"
                class="chakra-input css-15izoeg"
                data-testid="input-date"
                id="date"
                name="DATE"
                required=""
                type="time"
                value=""
              />
            </div>
            <div
              class="chakra-stack css-8g8ihq"
            >
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <label
                  class="chakra-form__label css-g6pte"
                  for="dropdown"
                  id="field-:r3:-label"
                >
                  DROPDOWN
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-1tfjd1n"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <div
                  class="chakra-select__wrapper css-1ylyrkq"
                >
                  <select
                    aria-required="true"
                    class="chakra-select css-16z42hf"
                    data-testid="input-dropdown"
                    id="dropdown"
                    name="DROPDOWN"
                    required=""
                  >
                    <option
                      value="Option 1"
                    >
                      Option 1
                    </option>
                    <option
                      value="Option 2"
                    >
                      Option 2
                    </option>
                  </select>
                  <div
                    class="chakra-select__icon-wrapper css-fcvg7i"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-select__icon"
                      focusable="false"
                      role="presentation"
                      style="width: 1em; height: 1em; color: currentColor;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <label
                class="chakra-form__label css-g6pte"
                for="message"
                id="field-:r4:-label"
              >
                MESSAGE
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-1tfjd1n"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <textarea
                aria-required="true"
                class="chakra-textarea css-45s8if"
                data-testid="input-message"
                id="message"
                name="MESSAGE"
                required=""
              />
            </div>
            <button
              class="chakra-button css-1vscnss"
              type="submit"
            >
              SUBMIT
            </button>
          </div>
        </form>
      </div>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>
  <div
    class="chakra-portal"
  >
    <div
      aria-label="Notifications-top"
      aria-live="polite"
      id="chakra-toast-manager-top"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <div
      aria-label="Notifications-top-left"
      aria-live="polite"
      id="chakra-toast-manager-top-left"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
    />
    <div
      aria-label="Notifications-top-right"
      aria-live="polite"
      id="chakra-toast-manager-top-right"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
    />
    <div
      aria-label="Notifications-bottom-left"
      aria-live="polite"
      id="chakra-toast-manager-bottom-left"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
    />
    <div
      aria-label="Notifications-bottom"
      aria-live="polite"
      id="chakra-toast-manager-bottom"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <div
      aria-label="Notifications-bottom-right"
      aria-live="polite"
      id="chakra-toast-manager-bottom-right"
      role="region"
      style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
    />
  </div>
</body>
`;

exports[`ContactForm match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div>
      <div
        class="css-1jlvcm7"
        id="contact"
      >
        <div
          class="css-b4kcmh"
        >
          <div
            class="css-ojqyia"
          >
            <img
              alt="Bg image - logo"
              class="chakra-image css-o0922y"
              src="https://sunkenstudio-strapi-cms.nyc3.digitaloceanspaces.com/b8c12199347804e795219ef484f5572e.jpeg"
              typename="ComponentCommonImage"
            />
            <div
              class="css-1bauqsl"
            />
          </div>
        </div>
        <div
          class="chakra-stack css-7pq1ey"
        >
          <h3
            class="chakra-heading css-qmgl8q"
          >
            Contact
          </h3>
          <form
            style="display: flex; width: 100%; justify-content: center;"
          >
            <div
              class="chakra-stack css-dpw1x7"
            >
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="name"
                  id="field-:r5:-label"
                >
                  NAME
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-0"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <input
                  aria-required="true"
                  class="chakra-input css-cxmp86"
                  data-testid="input-name"
                  id="name"
                  name="NAME"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="start_time"
                  id="field-:r6:-label"
                >
                  STARTTIME
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-0"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <input
                  aria-required="true"
                  class="chakra-input css-cxmp86"
                  data-testid="input-start_time"
                  id="start_time"
                  name="STARTTIME"
                  required=""
                  type="time"
                  value=""
                />
              </div>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="date"
                  id="field-:r7:-label"
                >
                  DATE
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-0"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <input
                  aria-required="true"
                  class="chakra-input css-cxmp86"
                  data-testid="input-date"
                  id="date"
                  name="DATE"
                  required=""
                  type="time"
                  value=""
                />
              </div>
              <div
                class="chakra-stack css-8g8ihq"
              >
                <div
                  class="chakra-form-control css-0"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-uvho5g"
                    for="dropdown"
                    id="field-:r8:-label"
                  >
                    DROPDOWN
                    <span
                      aria-hidden="true"
                      class="chakra-form__required-indicator css-0"
                      role="presentation"
                    >
                      *
                    </span>
                  </label>
                  <div
                    class="chakra-select__wrapper css-42b2qy"
                  >
                    <select
                      aria-required="true"
                      class="chakra-select css-1x2j97e"
                      data-testid="input-dropdown"
                      id="dropdown"
                      name="DROPDOWN"
                      required=""
                    >
                      <option
                        value="Option 1"
                      >
                        Option 1
                      </option>
                      <option
                        value="Option 2"
                      >
                        Option 2
                      </option>
                    </select>
                    <div
                      class="chakra-select__icon-wrapper css-162mkon"
                    >
                      <svg
                        aria-hidden="true"
                        class="chakra-select__icon"
                        focusable="false"
                        role="presentation"
                        style="width: 1em; height: 1em; color: currentColor;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="message"
                  id="field-:r9:-label"
                >
                  MESSAGE
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-0"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <textarea
                  aria-required="true"
                  class="chakra-textarea css-0"
                  data-testid="input-message"
                  id="message"
                  name="MESSAGE"
                  required=""
                />
              </div>
              <button
                class="chakra-button css-1qrv8y5"
                type="submit"
              >
                SUBMIT
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1jlvcm7"
      id="contact"
    >
      <div
        class="css-b4kcmh"
      >
        <div
          class="css-ojqyia"
        >
          <img
            alt="Bg image - logo"
            class="chakra-image css-o0922y"
            src="https://sunkenstudio-strapi-cms.nyc3.digitaloceanspaces.com/b8c12199347804e795219ef484f5572e.jpeg"
            typename="ComponentCommonImage"
          />
          <div
            class="css-1bauqsl"
          />
        </div>
      </div>
      <div
        class="chakra-stack css-7pq1ey"
      >
        <h3
          class="chakra-heading css-qmgl8q"
        >
          Contact
        </h3>
        <form
          style="display: flex; width: 100%; justify-content: center;"
        >
          <div
            class="chakra-stack css-dpw1x7"
          >
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="name"
                id="field-:r5:-label"
              >
                NAME
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <input
                aria-required="true"
                class="chakra-input css-cxmp86"
                data-testid="input-name"
                id="name"
                name="NAME"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="start_time"
                id="field-:r6:-label"
              >
                STARTTIME
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <input
                aria-required="true"
                class="chakra-input css-cxmp86"
                data-testid="input-start_time"
                id="start_time"
                name="STARTTIME"
                required=""
                type="time"
                value=""
              />
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="date"
                id="field-:r7:-label"
              >
                DATE
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <input
                aria-required="true"
                class="chakra-input css-cxmp86"
                data-testid="input-date"
                id="date"
                name="DATE"
                required=""
                type="time"
                value=""
              />
            </div>
            <div
              class="chakra-stack css-8g8ihq"
            >
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="dropdown"
                  id="field-:r8:-label"
                >
                  DROPDOWN
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-0"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <div
                  class="chakra-select__wrapper css-42b2qy"
                >
                  <select
                    aria-required="true"
                    class="chakra-select css-1x2j97e"
                    data-testid="input-dropdown"
                    id="dropdown"
                    name="DROPDOWN"
                    required=""
                  >
                    <option
                      value="Option 1"
                    >
                      Option 1
                    </option>
                    <option
                      value="Option 2"
                    >
                      Option 2
                    </option>
                  </select>
                  <div
                    class="chakra-select__icon-wrapper css-162mkon"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-select__icon"
                      focusable="false"
                      role="presentation"
                      style="width: 1em; height: 1em; color: currentColor;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="message"
                id="field-:r9:-label"
              >
                MESSAGE
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <textarea
                aria-required="true"
                class="chakra-textarea css-0"
                data-testid="input-message"
                id="message"
                name="MESSAGE"
                required=""
              />
            </div>
            <button
              class="chakra-button css-1qrv8y5"
              type="submit"
            >
              SUBMIT
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
